### ISSUE TITLE
User type needs to be linked with a default set of permissions

### ISSUE THREAD

**Issue by henokgetachew** [medic/medic-webapp#2951]
The whole point of having a user type seems to be that there are some default actions/permissions for that user type.

Example: The user type `Gateway` should have `can_access_gateway_api` permission turned on by default.

**Comment by garethbowen** [comment]
It's configured that way: https://github.com/medic/medic-webapp/blob/master/packages/kujua-sms/views/lib/app_settings.js#L190

@henokgetachew Did you upload any configurations after install or is this a clean default installation?

**Comment by henokgetachew** [comment]
@garethbowen this was actually when a 0.4 was ported to v2.  I see why it wasn't set by default. The app_settings uploaded belonged to the 0.4 originally.  I think this brings out one of the issues we will have to deal with when upgrading the instances to v2.  What's the best way to set these default values when app_settings has been uploaded from a previous instance? Tagging @abbyad too here.

**Comment by garethbowen** [comment]
Ahh right. What might be happening is any permissions that don't exist in the old configuration are left as null (false for everyone) in the new configuration. We might need to write some API code so add in missing translations with the default permissions.

**Comment by ngamita** [comment]
Yes, sure i agree @garethbowen - something like have defaults for v2 even though 0.4 configs are introduced taking care of the permissions that don't exist in the old configuration.

**Comment by abbyad** [comment]
The default permissions on an instance upgraded to v2.9 are:
```
  "permissions": [
    {
      "name": "can_export_messages",
      "roles": [
        "national_admin",
        "district_admin",
        "analytics"
      ]
    },
    {
      "name": "can_export_forms",
      "roles": [
        "national_admin",
        "district_admin",
        "analytics"
      ]
    },
    {
      "name": "can_export_contacts",
      "roles": [
        "national_admin",
        "district_admin"
      ]
    },
    {
      "name": "can_export_audit",
      "roles": [
        "national_admin"
      ]
    },
    {
      "name": "can_export_feedback",
      "roles": [
        "national_admin"
      ]
    },
    {
      "name": "can_export_server_logs",
      "roles": [
        "national_admin"
      ]
    },
    {
      "name": "can_access_directly",
      "roles": [
        "national_admin"
      ]
    },
    {
      "name": "can_view_analytics",
      "roles": [
        "national_admin",
        "district_admin",
        "analytics"
      ]
    },
    {
      "name": "can_view_data_records",
      "roles": [
        "national_admin",
        "district_admin",
        "analytics",
        "gateway"
      ]
    },
    {
      "name": "can_view_unallocated_data_records",
      "roles": [
        "national_admin",
        "district_admin",
        "gateway"
      ]
    },
    {
      "name": "can_edit",
      "roles": [
        "national_admin",
        "district_admin",
        "gateway"
      ]
    },
    {
      "name": "can_update_messages",
      "roles": [
        "national_admin",
        "district_admin",
        "gateway"
      ]
    },
    {
      "name": "can_create_records",
      "roles": [
        "national_admin",
        "district_admin",
        "date_entry",
        "gateway"
      ]
    },
    {
      "name": "can_view_tasks_tab",
      "roles": [
        "national_admin"
      ]
    },
    {
      "name": "can_configure",
      "roles": [
        "national_admin"
      ]
    },
    {
      "name": "can_view_messages_tab",
      "roles": [
        "national_admin",
        "district_admin"
      ]
    },
    {
      "name": "can_view_reports_tab",
      "roles": [
        "national_admin",
        "district_admin"
      ]
    },
    {
      "name": "can_view_analytics_tab",
      "roles": [
        "national_admin",
        "district_admin"
      ]
    },
    {
      "name": "can_view_contacts_tab",
      "roles": [
        "national_admin",
        "district_admin"
      ]
    }
```

**Comment by garethbowen** [comment]
It looks like this [migration runs](https://github.com/medic/medic-api/blob/master/migrations/add-permissions-configuration.js) which matches what @abbyad is seeing, and then no further permissions are added to the configuration.

**Comment by garethbowen** [comment]
PR please @SCdF 
* https://github.com/medic/medic-api/pull/110
* https://github.com/medic/medic-webapp/pull/2971

Defaults are now applied on API startup. In addition to defaulting undefined properties (as before) this also now iterates through the permissions array adding any that aren't found in the current settings.

Once merged I'll bump the `api` submodule and backport to `2.9.x`

**Comment by SCdF** [comment]
LGTM

**Comment by garethbowen** [comment]
Merged and backported to 2.9.x

**Comment by abbyad** [comment]
Moving to ready as I've seen the `can_access_gateway_api` included by default on upgrade from 0.4 to 2.9.0beta.9.

@henokgetachew if there are other settings that need to be added by default please open a new issue.
